packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -508,6 +512,12 @@ export class MirrorNodeClient {
       requestId?: string,
       searchableTypes: any[] = [constants.TYPE_CONTRACT, constants.TYPE_ACCOUNT, constants.TYPE_TOKEN]
     ) {
+        const cachedLabel = `resolveEntityType.${entityIdentifier}`;
+        const cachedResponse: { type: string, entity: any } | undefined = this.cache.get(cachedLabel);
+        if (cachedResponse != undefined) {
```

Comment:
Will this mean we'd cache an account that's not found to the pint that it could be created and we'd still return not found?

Let's add a test that follow the hollow account flow i.e. a check if done, the account doesn't exist, then the account is created, then a check is made and the account exists.